libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -120,9 +121,14 @@ public class BotFrameworkAdapter : BotAdapter, IAdapterIntegration, IExtendedUse
             IMiddleware middleware = null,
             ILogger logger = null)
         {
+            if (connectorClientRetryPolicy != null && customHttpClient != null)
+            {
+                throw new ArgumentException($"{nameof(connectorClientRetryPolicy)} should not be provided if {nameof(customHttpClient)} is provided");
+            }
+
             CredentialProvider = credentialProvider ?? throw new ArgumentNullException(nameof(credentialProvider));
             ChannelProvider = channelProvider;
-            _httpClient = customHttpClient ?? DefaultHttpClient;
+            _customHttpClient = customHttpClient;
```

Comment:
You do not need to provide a default client if I recall correctly. That is one point of this code change. The fact that a httpClient is set results in retry policy to not be honored. This is because the http client on which the retry policy is initialized is replaced, thus making retry to not work.